rust/private/stamp.bzl

```diff
@@ -0,0 +1,75 @@
+"""A small utility module dedicated to detecting whether or not the `--stamp` flag is enabled
```

Comment:
Could you please move this file into //rust/settings/?

Reply:
I feel strongly that this file should not need to exist in the first place and that the value of `--stamp` should be easily accessible within rules. https://github.com/bazelbuild/bazel/issues/11164 should be implemented and this file deleted so wherever it goes, I'd like it not to be a part of the public API.

Reply:
Sounds good.